File: spring-boot-s3-java-maven-aws-java-sdk/src/main/java/io/pivotal/tsalm/s3/infrastructure/config/InfrastructureConfiguration.java
package io.pivotal.tsalm.s3.infrastructure.config;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.s3.S3Client;

import java.net.URI;

@Configuration
class InfrastructureConfiguration {

    private S3ConfigurationProperties s3ConfigurationProperties;

    @Autowired
    public InfrastructureConfiguration(S3ConfigurationProperties s3ConfigurationProperties) {
        this.s3ConfigurationProperties = s3ConfigurationProperties;
    }

    @Bean
    public S3Client s3Client() {
        AwsBasicCredentials credentials = AwsBasicCredentials.create(s3ConfigurationProperties.getAccessKey(),
                s3ConfigurationProperties.getSecretKey());
        AwsCredentialsProvider credentialsProvider = StaticCredentialsProvider.create(credentials);
        return S3Client.builder()
                .credentialsProvider(credentialsProvider)
                .endpointOverride(URI.create(s3ConfigurationProperties.getEndpointUrl()))
                .region(Region.of(s3ConfigurationProperties.getRegion()))
                .build();
    }
}

File: spring-boot-s3-kotlin-gradle-aws-java-sdk/src/main/kotlin/io/pivotal/tsalm/s3/infrastructure/config/InfrastructureConfiguration.kt
package io.pivotal.tsalm.s3.infrastructure.config

import com.amazonaws.auth.AWSStaticCredentialsProvider
import com.amazonaws.auth.BasicAWSCredentials
import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration
import com.amazonaws.services.s3.AmazonS3
import com.amazonaws.services.s3.AmazonS3Client
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration


@Configuration
class InfrastructureConfiguration {

    @Value("\${s3Properties.endpointUrl}")
    private val endpointUrl: String? = null
    @Value("\${s3Properties.region}")
    private val region: String? = null
    @Value("\${s3Properties.accessKey}")
    private val accessKey: String? = null
    @Value("\${s3Properties.secretKey}")
    private val secretKey: String? = null

    @Bean
    fun amazonS3Client(): AmazonS3 {
        val credentialsProvider = AWSStaticCredentialsProvider(BasicAWSCredentials(accessKey, secretKey))
        return AmazonS3Client.builder()
                .withCredentials(credentialsProvider)
                .withEndpointConfiguration(EndpointConfiguration(endpointUrl, region))
                .build()
    }
}